packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -261,23 +261,56 @@ class AndroidSdk {
     return null;
   }
 
-  String getAvdManagerPath() {
-    final String binaryName = globals.platform.isWindows ? 'avdmanager.bat' : 'avdmanager';
-    final File cmdlineToolsBinary = directory
+  String getCmdlineToolsPath(String binaryName) {
+    // First look for the latest version of the command-line tools
+    final File cmdlineToolsLatestBinary = directory
       .childDirectory('cmdline-tools')
       .childDirectory('latest')
       .childDirectory('bin')
       .childFile(binaryName);
-    if (cmdlineToolsBinary.existsSync()) {
-      return cmdlineToolsBinary.path;
+    if (cmdlineToolsLatestBinary.existsSync()) {
+      return cmdlineToolsLatestBinary.path;
+    }
+
+    // Next look for the highest version of the command-line tools
+    final Directory cmdlineToolsDir = directory.childDirectory('cmdline-tools');
+    if (cmdlineToolsDir.existsSync()) {
+      final List<Version> cmdlineTools = cmdlineToolsDir
+        .listSync()
```

Comment:
Does this need to filter out files and only include Directories?

Reply:
Based on the [docs](https://developer.android.com/studio/command-line) I think it should only contain sub-directories.
All files will be discarded later anyways, although we could filter them out early just in case.